pkg/test/framework2/common/environment.go

```diff
@@ -0,0 +1,45 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package common
+
+// Keep the environment names here to avoid layering/circular dependency problems.
+
+const (
+	// Native environment name
+	Native EnvironmentName = "native"
```

Comment:
Seems pretty duplicative of framework2/components/environment/names.go. Seems like we would only need one publicly referencable const.

Reply:
There are a few tension points in the code due to the need for cross-references in multiple packages & layering. I created common to capture the settings/constants, but when writing user authored code (referencing environment names when skipping), it becomes awkward to reference common.<EnvName>.

Once I do the full porting, I'll take another stab at cleaning up the surface area.